src/Cardano/Wallet/Kernel/Wallet.hs

```diff
@@ -0,0 +1,119 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DuplicateRecordFields #-}
+{-# LANGUAGE ExistentialQuantification #-}
+{-# LANGUAGE TypeOperators #-}
+
+-- |
+-- Copyright: © 2018-2019 IOHK
+-- License: MIT
+--
+-- | Module provides internal implementation of
+-- | wallet api
+
+module Cardano.Wallet.Kernel.Wallet where
+
+import Prelude
+
+import Cardano.Wallet
+    ( Wallet )
+import Cardano.Wallet.AddressDerivation
+    ( Passphrase )
+import Cardano.Wallet.AddressDiscovery
+    ( AddressPoolGap )
+import Cardano.Wallet.Mnemonic
+    ( Mnemonic )
+import Cardano.Wallet.Primitive
+    ( Coin )
+import Crypto.Encoding.BIP39
+    ( ValidMnemonicSentence )
+import Data.Text
+    ( Text )
+import Data.Time.Units
+    ( Microsecond )
+import Fmt
+    ( Buildable (..), fmt )
+
+-- | Errors
+data CreateWalletError = CreateWalletError
+
+instance Buildable CreateWalletError where
+    build CreateWalletError =
+        mempty
+        <> "CreateWalletError"
+
+instance Show CreateWalletError where
+    show = fmt . build
+
+data GetWalletError = GetWalletError
+
+instance Buildable GetWalletError where
+    build GetWalletError =
+        mempty
+        <> "GetWalletError"
+
+instance Show GetWalletError where
+    show = fmt . build
```

Comment:
tip: When you're grooming an initial interface like this, you can leave this kind of stuff for later and focus on getting the types right first.

Reply:
yes, just wanted to signal that will be using `fmt` rather than `formatting`. Removed that